test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -885,6 +888,72 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
       });
     });
 
+    Expect.describe('Typing biometrics', function () {
+      itp('does not contain typing pattern header in primary auth request if feature is disabled', function () {
+        return setup({ features: { trackTypingPattern: false }})
+        .then(function (test) {
+          $.ajax.calls.reset();
+          test.form.setUsername('testuser');
+          test.form.setPassword('pass');
+          test.setNextResponse(resSuccess);
+          spyOn(TypingUtil, 'track').and.callFake(function (target) {
+            expect(target).toBe('okta-signin-username');
+          });
+          test.form.submit();
+          return tick();
+        })
+        .then(function () {
+          expect(TypingUtil.track).not.toHaveBeenCalled();
+          expect($.ajax.calls.count()).toBe(1);
+          var ajaxArgs = $.ajax.calls.argsFor(0);
+          expect(ajaxArgs[0].headers['X-Typing-Pattern']).toBe(undefined);
+        });
+      });
+
+      itp('contains typing pattern header in primary auth request if feature is enabled', function () {
+        return setup({ features: { trackTypingPattern: true }})
+        .then(function (test) {
+          $.ajax.calls.reset();
+          test.form.setUsername('testuser');
+          test.form.setPassword('pass');
+          test.setNextResponse(resSuccess);
+          spyOn(TypingUtil, 'track').and.callFake(function (target) {
+            expect(target).toBe('okta-signin-username');
+          });
+          spyOn(TypingUtil, 'getTypingPattern').and.callFake(function () {
+            return typingPattern;
+          });
+          test.form.submit();
+          return tick();
```

Comment:
same comment here about not using `tick` and use `Expect.waitForSpyCall` instead.
Same for other tests that are added